WalletWasabi.Gui/CommandLine/Daemon.cs

```diff
@@ -84,9 +86,21 @@ internal async Task RunAsync(string walletName, bool mixAll, bool keepMixAlive)
 					return;
 				}
 
-				await TryQueueCoinsToMixAsync(mixAll, password);
+				KeyManager destinationKeyManager = TryGetKeyManagerFromWalletName(destinationWalletName);
+				bool isDestinationSpecified = keyManager.ExtPubKey != destinationKeyManager.ExtPubKey;
+				if (isDestinationSpecified)
+				{
+					await Global.WalletManager.CreateAndStartWalletServiceAsync(destinationKeyManager);
+				}
+
+				// Enqueue coins up to the anonset target, except if mixall, because then enqueue all coins.
+				// However if destination is specified, then disregard this initial enqueue.
+				// This way mixall will have no effect. Or more specifically the output wallet specification will mix on top of mixed coins, so the result is the same.
+				if (!isDestinationSpecified)
+				{
+					await TryQueueCoinsToMixAsync(password, maxAnonset: mixAll ? int.MaxValue : WalletService.ServiceConfiguration.MixUntilAnonymitySet - 1);
+				}
```

Comment:
```suggestion
				// Enqueue coins up to the anonset target, except if mixall, because then enqueue all coins.
				// However if destination is specified, then disregard this initial enqueue.
				// This way mixall will have no effect. Or more specifically the output wallet specification will mix on top of mixed coins, so the result is the same.
				else
				{
					await TryQueueCoinsToMixAsync(password, maxAnonset: mixAll ? int.MaxValue : WalletService.ServiceConfiguration.MixUntilAnonymitySet - 1);
				}
```

Reply:
This is not equivalent.